htdocs/discover_tab.js

```diff
@@ -0,0 +1,58 @@
+var discover = function () {
+
+    /* "Model-Model-View-Controller"
+     .. or something ..
+     there are two models: the local state and the global rcs+github
+     the local model is private data of this class
+    */
+
+    // local model (all caches of stuff stored in RCS/github)
+    var notebook_info_ = {}, // all notebooks we are aware of
+        num_stars_ = {}, // number of stars for all known notebooks
+        fork_count_ = {};
+
+    function clean_r(obj) { 
+        delete obj.r_attributes; 
+        delete obj.r_type; 
+        return obj; 
+    };
+
+    function load_everything(){
+        var allnotebooks = rcloud.config.get_all_notebook_info();
+
+        var promises = [allnotebooks.then(function(notebooks) {
+            // populate notebook infos
+            notebook_info_ = clean_r(notebooks.infos);
+        }),
+        allnotebooks.then(function(notebooks) {
+            // populate star counts
+            num_stars_ = clean_r(notebooks.stars);
+        }),
+        allnotebooks.then(function(notebooks) {
+            // populate fork counts
+            fork_count_ = clean_r(notebooks.forks);
```

Comment:
`forks` doesn't exist in all_notebook_info, not that we should use it anyway. we'll probably need `rcloud.get_multiple_notebook_fork_counts` -- but that requires some server-side stuff. So if you'd rather not do that, @shaneporter, you could do something like 
```js
Object.keys(notebooks).map(function(id) { return rcloud.get_fork_count(id); })
```
for now. Slightly inefficient but shouldn't matter for 20 notebooks.